### ISSUE TITLE
GLideN64 Keeps Dumping Duplicate Textures

### ISSUE THREAD

**Issue by waltzz92** [gonetz/GLideN64#997]
Hi, I noticed that the same textures for a game keep getting dumped every time it shows up while playing. It seems the plugin does not check whether is has the texture dumped already. I have observed this in The Legend of Zelda Ocarina of Time, Majora's Mask, and Mario Party.

Steps to reproduce in Ocarina of Time:

1. Enable Texture Dumping on game start. Keep an eye on the dumped texture folder.
2. Observe the textures of Hyrule Field being dumped from the Title screen and when you start the game.
3. End emulation and close Project64 all the way.
4. Repeat steps 1-3. You should see the same textures being dumped, just with a different name.

This is easier to produce in Mario Party. If you play a Mini-Game while Texture Dumping is on, the game's textures are dumped every time you play the Mini-Game.

**Comment by loganmc10** [comment]
Does this happen with every game? I'm experiencing a memory leak on my Raspberry Pi when I play GoldenEye or Worms Armageddon, but not other games. The GPU memory is filling up with textures and eventually I run out of memory, I wonder if it is related to this

**Comment by waltzz92** [comment]
Yes, this happens with every game. When using GLideN64's texture dumping feature, textures are dumped every time they appear in-game, creating many duplicate textures. It makes it difficult to make my own HD Texture Packs.

**Comment by matthewharvey** [comment]
The different names might indicate that the textures have different CRC values, so the emulator is not counting them as the same texture.

**Comment by Sarkie** [comment]
I wonder if 
txHresAltCRC=1
Don't mind looking into this if I have a bit of time
would help with this and #1015 

Are the files exactly the same if you compare by CRC on disk?

**Comment by loganmc10** [comment]
So I tested this by enabling texture dumping on Worms Armageddon, and then doing an md5sum on all the dumped textures and looking for duplicates, there were lots.

Here are the md5sums:
[md5sums.txt](https://github.com/gonetz/GLideN64/files/307227/md5sums.txt)

Here is a list of the duplicate md5's:
[duplicates.txt](https://github.com/gonetz/GLideN64/files/307230/duplicates.txt)

Here is the textures I dumped:
[gliden.tar.gz](https://github.com/gonetz/GLideN64/files/307232/gliden.tar.gz)

**Comment by Sarkie** [comment]
Cool.

Using this 
http://www.freeware-guide.com/rareware/DupDetector.html

Shows quite a lot are the same or very similar images with 100% comparison with a 98% most are.

I wonder if just doing a single menu and dump textures, did W:A have lots of textures and moving screens? Just thinking if we have a simple example then compare the textures.

But I do think this is on the right tracks.

**Comment by loganmc10** [comment]
These textures were dumped during gameplay, the background is constantly changing (there are normally leaves blowing around and such), and the water is moving, but besides that it is mostly static.

**Comment by fzurita** [comment]
This could be a problem with GLideNHQ. Maybe this class: https://github.com/gonetz/GLideN64/blob/master/src/GLideNHQ/TxCache.cpp

**Comment by loganmc10** [comment]
@fzurita I normally compile with -DNOHQ=On, which disables all the GLideNHQ stuff, to perform this test I enabled GLideNHQ since it's the only way I can see to dump the textures from the cache. It would be nice if there way was a way to dump the "normal" texture cache.
